src/heap/gc-tracer.cc

```diff
@@ -393,6 +393,24 @@ void GCTracer::UpdateStatistics(GarbageCollector collector) {
     RecordGCSumCounters();
     combined_mark_compact_speed_cache_ = 0.0;
     long_task_stats->gc_full_atomic_wall_clock_duration_us += duration_us;
+    duration = duration + current_.incremental_marking_duration;
+    int64_t gced = static_cast<int64_t>(current_.start_object_size) - current_.end_object_size;
+    auto bad = MakeBytesAndDuration(gced, duration);
+    CHECK(!heap_->major_gc_bad);
+    heap_->major_gc_bad = bad;
+    if (has_last_gc) {
+      // assumption broken due to incremental gc
+      // CHECK(current_.start_object_size >= last_gc_end_bytes);
+      CHECK(current_.end_time - last_gc_end_time - duration > 0);
+      auto bad = MakeBytesAndDuration(std::max<int64_t>(0,
+                                                        current_.start_object_size - last_gc_end_bytes + heap_->AllocatedExternalMemorySinceMarkCompact()),
+                                      current_.end_time - last_gc_end_time - duration);
```

Comment:
Let's define `gced` and `duration` variables so that we don't need to split across lines.